### PR TITLE
Use key instead of pkhash

### PR DESCRIPTION
**Pull request by Kelvin-M**
The goal of this multi-repo PRs is to make sure that all the asset creation endpoints returns only the "key" of the asset. 
Indeed, currently `create` and 'get' an asset return, respectively, `pkhash` and `key`. As `pkhash` is only a substra-backend concept, we need to move from `pkhash` to `key` when dealing with substra cli/sdk and frontend.

Consequently, when client create a new asset, the substra-backend will now only return the `key` of the created asset.

There is an exception for compute plan which need to be returned in its complete form because client need to get the ID_to_key mapping and it’s returned only when client create this asset.

The client will take care of doing an extra GET in order to retrieve the complete asset.

PRs :
    https://github.com/SubstraFoundation/substra-backend/pull/312
    https://github.com/SubstraFoundation/substra/pull/220
    https://github.com/SubstraFoundation/substra-tests/pull/140

### CODE REVIEW

**Comment by natct10** [comment]
+ [documentation](https://github.com/SubstraFoundation/substra-documentation/pull/110)

File: substra/sdk/backends/remote/rest_client.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _add(self, name, exist_ok=False, **request_kwargs):
             if not exist_ok:
                 raise
 
-            key = e.pkhash or e.computePlanID
+            key = e.key or e.computePlanID
```
Comment (Esadruhn) [inline substra/sdk/backends/remote/rest_client.py:215]: actually e does not have a 'computePlanID' attribute, you can directly use `e.key`

Comment (Kelvin-M) [reply substra/sdk/backends/remote/rest_client.py:215]: You're right (https://github.com/SubstraFoundation/substra/blob/bb0cf81b237d6e3f20a0dda034141193fa4db20d/substra/sdk/exceptions.py#L113-L117)

**Comment by Esadruhn** [comment]
https://github.com/SubstraFoundation/substra/pull/220/files#diff-434b67ddd91e180773c318731c47bcfdR138

Is removing the call to `get` after the `add` part of this MR ? Or is it OK if I commit on this MR ?

**Comment by Esadruhn** [comment]
> > https://github.com/SubstraFoundation/substra/pull/220/files#diff-434b67ddd91e180773c318731c47bcfdR138
> > Is removing the call to `get` after the `add` part of this MR ? Or is it OK if I commit on this MR ?
> 
> @Esadruhn You can commit in the PR for this part 👍

It is a lot of changes in the end (with the choice of getting either the key or asset) so I put it in a separate MR: #221

**Review by Esadruhn — APPROVED** [review]
LGTM !

File: substra/sdk/backends/remote/rest_client.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def _add(self, name, exist_ok=False, **request_kwargs):
             if not exist_ok:
                 raise
 
-            key = e.pkhash or e.computePlanID
+            key = e.key
```
Comment (jmorel) [inline substra/sdk/backends/remote/rest_client.py:215]: Are you sure you can remove handling the computePlanID here? Are we sure there cannot be an "AlreadyExists" exception when creating a compute plan?

File: substra/sdk/backends/remote/rest_client.py (hunk 1 above)
Comment (Kelvin-M) [reply substra/sdk/backends/remote/rest_client.py]: @jmorel

File: substra/sdk/backends/remote/rest_client.py (hunk 2 above)
Comment (Kelvin-M) [reply substra/sdk/backends/remote/rest_client.py:215]: See there : https://github.com/SubstraFoundation/substra/pull/220#discussion_r482822939

**Comment by Kelvin-M** [comment]
Need review on this one : @jmorel @inalgnu @samlesu

**Review by jmorel — APPROVED** [review]

**Review by inalgnu — APPROVED** [review]

**Pull request merged**